Clean up setup.py

ttung opened the pull request:
# Description
1. napari-gui is gone.
2. collapse comment removal from requirements.txt into one place.
3. [setup()](https://packaging.python.org/guides/distributing-packages-using-setuptools/#setup-args) doesn't have a `requires` kwarg.
4. use [literal string concatenation](https://docs.python.org/3/reference/lexical_analysis.html#string-literal-concatenation).  IDEs know how to deal with literal string concatenation better.
5. move the block of string constants after the imports for PEP8 compliance.

## Type of change
<!-- Please delete options that are not relevant. -->
- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?
<!-- Please describe the tests that you ran to verify your changes. -->
```
mkdir /tmp/i
cd /tmp/i
python3.6 venv .venv
pip install -e ~/imaging/napari
```

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

sofroniewn reviewed (commented):
This looks good to me, though maybe @kne42 can take a look too. I think we'll still want to merge #888 and #887 first (which may cause a tiny merge conflict)

jni commented on setup.py:
```diff
@@ -4,6 +4,13 @@
 GUI component of Napari.
```
Might as well update this docstring? =)

ttung replied:
Done.

kne42 commented on setup.py:
```diff
@@ -1,9 +1,14 @@
-"""Napari GUI
-
-GUI component of Napari.
+"""Napari Viewer
```
could you replace this with a subset of our README? it will be used as part of the pip description

ttung replied:
Done.

kne42 commented on setup.py:
```diff
@@ -34,61 +39,33 @@
-    sep = l.split(' #')
-    INSTALL_REQUIRES.append(sep[0].strip())
-    if len(sep) == 2:
-        REQUIRES.append(sep[1].strip())
```
idk why everyone removed the comments from the requirements file :(

ttung replied:
I re-added the logic to strip comments, but integrated the comment-at-start-of-line and the comment-at-end-of-line logic into one thing.

kne42 reviewed (commented):
Thanks for the contribution @ttung!

I'm confused why you removed the `packages` kwarg; it's listed in the link you provided ([link to `packages` section](https://packaging.python.org/guides/distributing-packages-using-setuptools/#packages)). Has it been made optional or something?

tlambert03 commented on setup.py:
```diff
@@ -39,52 +48,37 @@
 ]
 
+
 if sys.version_info < (3, 6):
```
could we get `MIN_PY_VER` in here?

ttung replied:
Updated.

sofroniewn reviewed (approved):
Thanks for the updates @ttung - this looks good to me. I will let sit for a bit in case anyone else wants to weigh in and then merge.

Commit:
Clean up setup.py

1. napari-gui is gone.
2. collapse comment removal from requirements.txt into one place.
3. [setup()](https://packaging.python.org/guides/distributing-packages-using-setuptools/#setup-args) doesn't have a `requires` kwarg.
4. use [literal string concatenation](https://docs.python.org/3/reference/lexical_analysis.html#string-literal-concatenation).  IDEs know how to deal with literal string concatenation better.
5. move the block of string constants after the imports for PEP8 compliance.
